homeassistant/components/mqtt/device_trigger.py

```diff
@@ -267,12 +301,44 @@ async def async_setup_trigger(
 ) -> None:
     """Set up the MQTT device trigger."""
     config = TRIGGER_DISCOVERY_SCHEMA(config)
-    device_id = update_device(hass, config_entry, config)
+
+    # Check if there is an existing conflicting trigger first
+    device_registry = dr.async_get(hass)
+    device_entry = device_registry.async_get_device(
+        identifiers={(DOMAIN, id_) for id_ in config[CONF_DEVICE][CONF_IDENTIFIERS]},
+        connections={
+            (conn_[0], conn_[1]) for conn_ in config[CONF_DEVICE][CONF_CONNECTIONS]
+        },
+    )
+    if device_entry is not None:
+        device_id = device_entry.id
+        discovery_id = discovery_data[ATTR_DISCOVERY_HASH][1]
+        trigger_type = config[CONF_TYPE]
+        trigger_subtype = config[CONF_SUBTYPE]
+        mqtt_data = get_mqtt_data(hass)
+        for _, trig in mqtt_data.device_triggers.items():
+            if (
+                trig.discovery_data is not None
+                and trig.device_id == device_id
+                and trig.type == trigger_type
+                and trig.subtype == trigger_subtype
+            ):
+                _LOGGER.error(
+                    "Config for device trigger %s conflicts with existing "
+                    "device trigger, cannot set up trigger, got: %s",
+                    discovery_id,
+                    config,
+                )
+                send_discovery_done(hass, discovery_data)
+                clear_discovery_hash(hass, discovery_data[ATTR_DISCOVERY_HASH])
+                return None
+
+    new_device_id = update_device(hass, config_entry, config)
 
```

Comment:
Before we set up an MQTT device trigger, we check if there is an existing trigger with the same `discovery_id` already. In that case we abort the discovery. The device trigger created will remain inactive till it is cleaned up. This is tested too.
May be we should initiate that the trigger should be removed too, not just the discovery hash.